source/module_io/numerical_basis.cpp

```diff
@@ -122,7 +125,38 @@ void Numerical_Basis::output_overlap(const psi::Psi<std::complex<double>>& psi,
             // (2) generate Sq matrix if necessary.
             if (winput::out_spillage == 2)
             {
+#ifndef __LCAO
+                // compute <jY|jY> in plane-wave basis
                 overlap_Sq[ik] = this->cal_overlap_Sq( ik, npw, static_cast<double>(derivative_order), sf, wfcpw);
+#else
+                // compute <jY|jY> with two-center integration
+                assert(derivative_order == 0 || derivative_order == 1);
+                char type = (derivative_order == 0) ? 'S' : 'T';
+                std::vector<int> natom;
+                std::vector<int> lmax;
+                std::vector<std::vector<ModuleBase::Vector3<double>>> tau_cart;
+                for (int it = 0; it < GlobalC::ucell.ntype; ++it)
```

Comment:
I would suggest passing GlobalC::ucell into this function instead of directly using GlobalC

Reply:
I definitely agree with this idea! The latest commit changes the all usage of ucell within the NumericalBasis class from GlobalC to function arguments.